ITensorGaussianMPS/src/gmps.jl

```diff
@@ -391,6 +844,116 @@ function interleave(xs...)
   return res
 end
 
+function interleave(a::ConservesNf{T}, b::ConservesNf{T}) where {T}
+  return set_data(a, interleave(a.data, b.data))
+end
+function interleave(a::ConservesNfParity{T}, b::ConservesNfParity{T}) where {T}
+  return set_data(
+    a,
+    interleave(
+      interleave(a.data[1:2:end], b.data[1:2:end]),
+      interleave(a.data[2:2:end], b.data[2:2:end]),
+    ),
+  )
+end
+
+function interleave(M::AbstractMatrix)
+  @assert size(M, 1) == size(M, 2)
+  n = div(size(M, 1), 2)
+  first_half = Vector(1:n)
+  second_half = Vector((n + 1):(2 * n))
+  interleaved_inds = interleave(first_half, second_half)
+  return M[interleaved_inds, interleaved_inds]
+end
+
+function interleave(g1::Circuit, g2::Circuit)
+  return Circuit(interleave(g1.rotations, g2.rotations))
+end
+
+function reverse_interleave(M::AbstractMatrix)
+  @assert size(M, 1) == size(M, 2)
+  n = div(size(M, 1), 2)
+  first_half = Vector(1:n)
+  second_half = Vector((n + 1):(2 * n))
+  interleaved_inds = interleave(first_half, second_half)
+  ordered_inds = sortperm(interleaved_inds)
+  return M[ordered_inds, ordered_inds]
+end
+
+function correlation_matrix_to_mps(
+  s::Vector{<:Index},
+  Λ_up0::AbstractSymmetry,
+  Λ_dn0::AbstractSymmetry;
```

Comment:
Dispatching on number of sites being a (1,2 x) multiple of the size of the correlation matrix like in the spinless case is non-unique here. For example s could be 2x the size of the correlation matrix if two species of spinless `Fermion` is used and the state is number conserving, or it could be 2x the size of the correlation matrix if `Electron` sites are used and the state is only parity conserving. Possible fix could be dispatching on sitetypes ( `"Electron"` , `"Fermion"`)?

Reply:
Maybe you figured it out, but yes I had been dispatching on the site types if I remember correctly.